phoneApp/js/router.js

```diff
@@ -0,0 +1,97 @@
+class Router {
+//	constructor(activePage) {
+//		this.activePage = activePage;
+	constructor(globalState) {
+		this.state = globalState;
+	}
+
+	initializeRouter() {
+		const mountNode = document.getElementById("mountNode");
+		//console.log(mountNode);
+		mountNode.innerHTML = `
+        <div id="app"></div>
+        <footer class="footer">
+            <div class="container bottom-radius">
+                <nav class="main-nav">
+		            ${this.renderLink({
+						href: "contacts",
+						glyphicon: "search",
+						text: "Contacts",
+						active: true
+					})}
+		            ${this.renderLink({
+						href: "keypad",
+						glyphicon: "th",
+						text: "Keypad",
+						active: false
+					})}
+		            ${this.renderLink({
+						href: "editcontact",
+						glyphicon: "pencil",
+						text: "Edit contact",
+						active: false
+					})}
+		            ${this.renderLink({
+						href: "user",
+						glyphicon: "user",
+						text: "User",
+						active: false
+					})}
+		            ${this.renderLink({
+						href: "adduser",
+						glyphicon: "plus",
+						text: "Add user",
+						active: false
+					})}    
+                 </nav>
+             </div>
+         </footer>`;
+
+		this.linksParent = document.querySelector(".main-nav");
+		//this.state.appDOMNode = document.getElementById("app"); // сюда будем делать рендер всех страниц
+		// и это не будет затрагивать футер и его события
+	}
+
+	switchRouter() {
+		this.linksParent.addEventListener("click", e => {
+			e.preventDefault();
+			//let target = e && e.target;
+			let target =
+				e &&
+				e.target &&
+				(e.target.closest("a") || e.target.classList.contains("tab")); //You can add additional attributes
```

Comment:
I'm pretty sure you don't need such event delegation because of your event working on a small area of responsibilities.

function switchRounter, should only change the router and don't make any checks and probably has a minimal of conditionalsa